storage/sql_test.go

```diff
@@ -0,0 +1,423 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/go-pg/pg/v10"
+	"github.com/stretchr/testify/assert"
+)
+
+// may want to change this to use a named schema such as lotus
+const PGSchema = "public"
+
+// resetSchema ensures the database schema is clean for a test
+func resetSchema(db *Database) error {
+	_, err := db.DB.Exec(fmt.Sprintf(`DROP SCHEMA %s CASCADE`, PGSchema))
+	if err != nil {
+		return err
+	}
+
+	_, err = db.DB.Exec(fmt.Sprintf(`CREATE SCHEMA %s`, PGSchema))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type column struct {
+	name     string
+	datatype string
+	nullable bool
+}
+
+var expectedTables = []struct {
```

Comment:
passing comment: as the number of tables grow I think this pattern may become tedious to maintain.

Reply:
I can see that, but I think it's valuable as a guard against regressions and as a prompt to write a migration.